test/integration/roundtrip_test.go

```diff
@@ -0,0 +1,227 @@
+	if len(brokers.Items) > 0 {
+		t.Fatalf("brokers should not exist on start, had %v brokers", len(brokers.Items))
+	}
+
+	brokerServer, err := brokerClient.Create(broker)
+	if nil != err {
+		t.Fatal("error creating the broker", broker)
+	}
+	if broker.Name != brokerServer.Name {
+		t.Fatalf("didn't get the same broker back from the server \n%+v\n%+v", broker, brokerServer)
+	}
+
+	brokers, err = brokerClient.List(v1.ListOptions{})
+	if 1 != len(brokers.Items) {
+		t.Fatalf("should have exactly one broker, had %v brokers", len(brokers.Items))
+	}
+
+	brokerServer, err = brokerClient.Get(broker.Name)
+	if broker.Name != brokerServer.Name &&
+		broker.ResourceVersion == brokerServer.ResourceVersion {
+		t.Fatalf("didn't get the same broker back from the server \n%+v\n%+v", broker, brokerServer)
+	}
+
+	// check that the broker is the same both ways
+	brokerListed := &brokers.Items[0]
+	if !reflect.DeepEqual(brokerServer, brokerListed) {
+		t.Fatalf("didn't get the same broker twice", brokerServer, brokerListed)
+	}
+
+	brokerServer.Spec.AuthUsername = "dug"
+	brokerServer.Spec.AuthPassword = "paul"
+	brokerUpdated, err := brokerClient.Update(brokerServer)
+	if nil != err ||
+		"dug" != brokerUpdated.Spec.AuthUsername ||
+		"paul" != brokerUpdated.Spec.AuthPassword {
+		t.Fatal("broker wasn't updated", brokerServer, brokerUpdated)
+	}
+
+	brokerServer, err = brokerClient.Get("test-broker")
+	if nil != err ||
+		"dug" != brokerServer.Spec.AuthUsername ||
+		"paul" != brokerServer.Spec.AuthPassword {
+		t.Fatal("broker wasn't updated", brokerServer)
+	}
+
+	err = brokerClient.Delete("test-broker", &v1.DeleteOptions{})
+	if nil != err {
+		t.Fatal("broker should be deleted", err)
+	}
+
+	brokerDeleted, err := brokerClient.Get("test-broker")
+	if nil == err {
+		t.Fatal("broker should be deleted", brokerDeleted)
+	}
+}
+
+func getFreshApiserverAndClient(t *testing.T) (servicecatalogclient.Interface, func()) {
+	securePort := 65535
+	serverIP := fmt.Sprintf("https://localhost:%d", securePort)
+	stopCh := make(chan struct{})
+	serverFailed := make(chan struct{})
+	//defer close(stopCh)
+	shutdown := func() {
+		close(stopCh)
+	}
+
+	secureServingOptions := genericserveroptions.NewSecureServingOptions()
+	go func() {
+		options := &server.ServiceCatalogServerOptions{
+			GenericServerRunOptions: genericserveroptions.NewServerRunOptions(),
+			SecureServingOptions:    secureServingOptions,
+			EtcdOptions:             genericserveroptions.NewEtcdOptions(),
+			AuthenticationOptions:   genericserveroptions.NewDelegatingAuthenticationOptions(),
+			AuthorizationOptions:    genericserveroptions.NewDelegatingAuthorizationOptions(),
+		}
+		options.SecureServingOptions.ServingOptions.BindPort = securePort
+		options.EtcdOptions.StorageConfig.ServerList = []string{"http://localhost:2379"}
+		if err := options.RunServer(stopCh); err != nil {
+			close(serverFailed)
+			t.Fatalf("Error in bringing up the server: %v", err)
```

Comment:
Just some nitpicking: this should probably be:

```
fmt.Errorf("Error in bringing up the server: %v", err)
close(serverFailed)
```

`t.Fatalf` calls `t.FailNow`, which doesn't work if called in a different goroutine (which is why I'm assuming you have the `serverFailed` channel in the first place; otherwise I wouldn't think it'd be necessary).